rfc/rfc-59/rfc-59.md

```diff
@@ -0,0 +1,285 @@
+
+Firstly, the `preCombineWithMultipleFields` function helps Hudi to deduplicate records inside a buffered bucket with same primary key(s) by multiple event-time fields and only remain the latest record whose every event-time field is latest. 
+
+This function would be invoked in `hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/commit/FlinkWriteHelper.java` # `deduplicateRecordsWithMultipleCombineFields` to deduplicate record. The deduplication process will also be described later.
+
+Secondly, the `needUpdatingPersistedRecordWithMultipleFields` functions helps Hudi to verify whether all event-time fields are the latest compared with the record persisted on the Disk when sinking the deduplicated record to the Disk before real write operation conducts.
+
+The high-level entire design of this Payload class is shown here:
+
+```java
+
+// hudi-common/src/main/java/org/apache/hudi/common/model/VerifyMultipleLatestEventTimesPayload.java
+public class VerifyMultipleLatestEventTimesPayload extends BaseAvroPayload implements HoodieRecordPayload<VerifyMultipleLatestEventTimesPayload> {
+
+  // A HashMap in order to store the relationship between event-time fieldnames and values
+  private HashMap<String, Comparable<?>> eventTimeFieldNamesAndValues;
+
+  // A string in order to save the eventime-time fieldname of orderingVal. 
+  // This is a compromise designed to be compatible with the current single event-time (property 'orderingVal' provided by BaseAvroPayload). 
+  private String orderingValFieldName;
+
+  // Construction compatible with the current workflow
+  public VerifyMultipleLatestEventTimesPayload(GenericRecord record, Comparable orderingVal, String orderingValFieldName) ;
+
+  // Construction for the multiple event-time latest verification
+  public VerifyMultipleLatestEventTimesPayload(GenericRecord record, HashMap<String, Comparable<?>> eventTimeFieldNamesAndValues) ;
+
+  /**
+  * It helps Hudi to deduplicate records inside a buffered bucket with same primary key(s) by multiple event-time fields and only remain the latest record whose every event-time field is the latest.
+  * About the function name, instead of precombine, we believe that use a different function name is better. 
+  * But it could also be named as preCombine in the future if we finished merging the workflows of single and multiple event-times processing
+  *
+  * @param previousValue another instance of {@link VerifyMultipleLatestEventTimesPayload} which needs to be compared with this one and choose the one whose every event-time is the latest (null value should be dealt carefully)
+  * @returns latestPayload return the chosen recordPayload (every event-time field is latest)
+  */
+  public VerifyMultipleLatestEventTimesPayload preCombineWithMultipleFields(VerifyMultipleLatestEventTimesPayload previousValue) ;

```

Comment:
let's not add new methods. we can add new overloaded methods. but not new methods w/ new names. It achieves the same purpose as existing method just that instead of 1 ordering field, its now N fields.